test/compilable/traits.d

```diff
@@ -13,8 +13,16 @@ class C19152
 }
 
 static assert(is(typeof(__traits(getTargetInfo, "cppRuntimeLibrary")) == string));
-
 version (CppRuntime_Microsoft)
 {
     static assert(__traits(getTargetInfo, "cppRuntimeLibrary") == "libcmt");
 }
+
+static assert(__traits(getTargetInfo, "floatAbi") == "hard");
```

Comment:
version (D_HardFloat)

Reply:
There is no DMD build where this is not true.

Reply:
How do you think gdc and ldc test their compilers?

Reply:
Really? I'd be astonished if the DMD test suite ran on all platforms/targets unmodified O_o

Reply:
LDC and GDC each use different build systems, but apart from that they all use dmd, druntime and phobos's test suites, in addition to their own test suites. Very often you can see PRs from GDC or LDC to upstream fixing bugs in some tests, exactly because they've run the tests on platforms that dmd doesn't support, where the assumptions doesn't hold, or because the test was relying on undefined behavior which got stressed by their backend optimizers in a different way.

Reply:
The less modifications I have to maintain, the better. Ideally the D2 testsuite would just be copied verbatim.

After all, it should be representative of what a *conforming* D compiler should be. :-)